hydroflow/tests/surface_async.rs

```diff
@@ -1,9 +1,218 @@
+//! Surface syntax tests of asynchrony and networking.
+
 use std::cell::RefCell;
+use std::collections::HashSet;
 use std::error::Error;
+use std::net::{Ipv4Addr, SocketAddr};
 use std::rc::Rc;
 use std::time::Duration;
 
-use hydroflow::hydroflow_syntax;
+use hydroflow::scheduled::graph::Hydroflow;
+use hydroflow::{hydroflow_syntax, rassert, rassert_eq};
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::net::{TcpListener, TcpStream, UdpSocket};
+use tokio::task::LocalSet;
+use tokio_stream::wrappers::LinesStream;
+
+#[tokio::test]
+pub async fn test_echo_udp() -> Result<(), Box<dyn Error>> {
+    const SERVER_PORT: u16 = 8099;
+    let server_socket = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), SERVER_PORT);
+
+    let local = LocalSet::new();
+
+    // Server:
+    let serv = local.spawn_local(async {
+        let socket = UdpSocket::bind((Ipv4Addr::UNSPECIFIED, SERVER_PORT))
+            .await
+            .unwrap();
+        let (send, recv) = hydroflow::util::udp_lines(socket);
+        println!("Server live!");
+
+        let seen = <Rc<RefCell<HashSet<String>>>>::default();
+        let seen_inner = Rc::clone(&seen);
+
+        let mut df: Hydroflow = hydroflow_syntax! {
+            recv_stream(recv)
+                -> map(|r| r.unwrap())
+                -> map(|x| { seen_inner.borrow_mut().insert(x.0.clone()); x })
+                -> sink_async(send);
+        };
+
+        tokio::select! {
+            _ = df.run_async() => (),
+            _ = tokio::time::sleep(Duration::from_secs(1)) => (),
+        };
+
+        rassert_eq!(4, seen.borrow().len())?;
+        rassert!(seen.borrow().contains("Hello"))?;
+        rassert!(seen.borrow().contains("World"))?;
+        rassert!(seen.borrow().contains("Raise"))?;
+        rassert!(seen.borrow().contains("Count"))?;
+
+        Ok(()) as Result<(), Box<dyn Error>>
+    });
+
+    // Client A:
+    let client_a = local.spawn_local(async move {
+        let socket = UdpSocket::bind((Ipv4Addr::UNSPECIFIED, 0))
+            .await
+            .unwrap();
+        let (send, recv) = hydroflow::util::udp_lines(socket);
+
+        let seen = <Rc<RefCell<Vec<String>>>>::default();
+        let seen_inner = Rc::clone(&seen);
```

Comment:
same questions as above about using join for a more organic Hydroflow script for the test.